### ISSUE TITLE
False positive when validating a CSS variable whose name includes a hyphen + integer and is not the first line

### ISSUE THREAD

**Issue by fengtan** [squizlabs/PHP_CodeSniffer#3364]
**Describe the bug**
When validating a CSS file that contains a [variable (aka custom property)](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) and:
* The variable name includes an hyphen followed by an integer
* The variable is not defined on the first line inside the class

Then the error `Style definitions must end with a semicolon` from [Squiz.CSS.SemicolonSpacingSniff](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Standards/Squiz/Sniffs/CSS/SemicolonSpacingSniff.php) shows up.

**Code sample**
```css
:root {
    --foo: #000;
    --bar-1: #000; /* Leads to false positive */
}
```

**Custom ruleset**
Not applicable

**To reproduce**
1. Create a file called `test.css` with the code sample above
2. Run `phpcs --standard=Squiz test.css`
3. See error message displayed
```
FILE: test.css
----------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
----------------------------------------------------------------------
 3 | ERROR | Style definitions must end with a semicolon
----------------------------------------------------------------------
```

**Expected behavior**
There should be no error.

CSS variables are allowed to include hyphens and integers. While the [documentation](https://developer.mozilla.org/en-US/docs/Web/CSS/--*) does not explicitly state that integers are allowed in property names, this is a common practice (example: https://www.smashingmagazine.com/2018/05/css-custom-properties-strategy-guide/).

**Versions**
 - OS: Debian GNU/Linux 10 (buster)
 - PHP: 7.4.16
 - PHPCS: 3.6.0
 - Standard: Squiz

**Additional context**
:heavy_check_mark: This works (no line before the problematic line):
```
:root {
    --bar-1: #000;
}
```
:heavy_check_mark: This works (additional line *after*, not before, the problematic line):
```
:root {
    --bar-1: #000;
    --foo: #000;
}
```
:heavy_check_mark: This works (no hyphen before the integer):
```
:root {
    --foo: #000;
    --bar1: #000;
}
```
The problem may lie in the [CSS tokenizer](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Tokenizers/CSS.php), or rather in the [PHP tokenizer](https://github.com/squizlabs/PHP_CodeSniffer/blob/master/src/Tokenizers/PHP.php) (from which the former inherits).

**Comment by gsherwood** [comment]
Support for checking JS and CSS files is being removed in version 4, and I have stopped supporting these features in version 3, so this issue will not be fixed. I strongly recommend you use a different tool for linting your CSS files.
